WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+        
+        if let searchText = searchText where searchText.characters.count > 0 {
+            let searchPredicate = NSPredicate(format: "postTitle CONTAINS[cd] %@", searchText)
+            predicates.append(searchPredicate)
+        }
+        
+        let predicate = NSCompoundPredicate(andPredicateWithSubpredicates: predicates)
+        return predicate
+    }
+    
+    // MARK: - Table View Handling
+    
+    private func tableView(tableView: UITableView, estimatedHeightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as! Post
+        
+        if cellIdentifierForPost(post) == self.dynamicType.postCardRestoreCellIdentifier {
+            return CGFloat(self.dynamicType.postCardRestoreCellRowHeight)
+        }
+        
+        return CGFloat(self.dynamicType.postCardEstimatedRowHeight)
+    }
+    
+    private func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let width = CGRectGetWidth(tableView.bounds)
+        return self.tableView(tableView, heightForRowAtIndexPath: indexPath, forWidth: width)
+    }
+   
+    private func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath, forWidth width: CGFloat) -> CGFloat {
+        guard let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? Post else {
+            return 0
+        }
+        
+        if cellIdentifierForPost(post) == self.dynamicType.postCardRestoreCellIdentifier {
+            return CGFloat(self.dynamicType.postCardRestoreCellRowHeight)
+        }
+        
+        var cell : PostCardTableViewCell!
+        
+        if post.pathForDisplayImage?.characters.count > 0 {
+            cell = textCellForLayout
+        } else {
+            cell = imageCellForLayout
+        }
+        
+        configureCell(cell, atIndexPath: indexPath)
+        let size = cell.sizeThatFits(CGSizeMake(width, CGFloat.max))
+        let height = ceil(size.height)
+        
+        return height
+    }
+    
+    private func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        tableView.deselectRowAtIndexPath(indexPath, animated: true)
+        
+        guard let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? AbstractPost else {
+            return
+        }
+
+        if post.remoteStatus == AbstractPostRemoteStatusPushing {
+            // Don't allow editing while pushing changes
+            return
+        }
+        
+        if post.status == PostStatusTrash {
+            // No editing posts that are trashed.
+            return
+        }
+        
+        previewEditPost(post)
+    }
+    
+    private func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as! Post
```

Comment:
If this VC is trying to show anything other than posts, meaning there's some serious invalid state inside it - that's a good enough reason to shut down the app to prevent further data corruption.  There's no telling what else can break if we try to pretend nothing's wrong.

Also - there's no telling what kind of invalid bug reports we'll get if we try to pretend there's no issue here - as opposed to crashing right here, in which case the stack traces are going to show exactly what the problem is.

We should make sure we crash sooner rather than later, in a controlled way.  Invalid state is evil and should not be hidden.